mitodl/ocw-studio
# Codify new mass publish pipeline and api endpoint

#### Pre-Flight checklist


#### What's this PR do?
Closes #926

#### How should this be manually tested?
Start containers with `docker-compose --profile concourse up` (make sure you have the required settings values in .env as detailed in the README).
Run `manage.py upsert_mass_publish_pipeline`, it should work and there should be `mass-publish/version:live` and `mass-publish/version:draft` pipelines in concourse
Run `manage.py upsert_theme_assets_pipeline`, it should still work
Run `manage.py mass_publish <with optional filters> draft`, it should work, the publish statuses for the filtered sites should be set to 'not started', and there should be one job triggered for  "mass-publish/version:draft" 

Using curl or postman, you can go to `http://localhost:8043/api/publish/?version=<draft or live>` with an authorization header of `Bearer <settings.API_BEARER_TOKEN>` to see what sites are being supplied to the pipeline.

Create a new site, make some content, and publish.  There should be a pipeline created for it (live and draft) and the appropriate version-specific one should be triggered when you publish.
